server/api/cli/cli.controller.js

```diff
@@ -39,17 +39,22 @@ exports.run = function (req, res) {
   dirName += dateTime.getFullYear()
   const execFile = req.body.fileName.replace('.c', '')
   const cmd = `"${dirName}/${execFile}"`
-  exec(cmd, { timeout: 10000 }, // Process will time out if running for > 10 seconds.
+  var cp = exec(cmd, { timeout: 10000 }, // Process will time out if running for > 10 seconds.
     (error, stdout, stderr) => {
       if (error) {
-        return res.send(200, error)
+        return res.send(200, {"info":error, "process":cp.pid});
```

Comment:
I'll try writing a test with Express later today. But if you are looking now, I would guess it has something to do with a promise. If Express doesn't wait for the promise to resolve that would explain why it returns a response before the timeout.